gobblin-modules/gobblin-kafka-09/src/test/java/org/apache/gobblin/runtime/HighLevelConsumerTest.java

```diff
@@ -176,6 +177,33 @@ public void testCalculateProduceToConsumeLag() {
     Assert.assertTrue(consumer.calcMillisSince(produceTimestamp).equals(234L));
   }
 
+  @Test
+  public void testQueueProcessorRuntimeExceptionEncountered() throws Exception {
+    Properties consumerProps = new Properties();
+    consumerProps.setProperty(ConfigurationKeys.KAFKA_BROKERS, _kafkaBrokers);
+    consumerProps.setProperty(Kafka09ConsumerClient.GOBBLIN_CONFIG_VALUE_DESERIALIZER_CLASS_KEY, "org.apache.kafka.common.serialization.ByteArrayDeserializer");
+    consumerProps.setProperty(SOURCE_KAFKA_CONSUMERCONFIG_KEY_WITH_DOT + KAFKA_AUTO_OFFSET_RESET_KEY, "earliest");
+    //Generate a brand new consumer group id to ensure there are no previously committed offsets for this group id
+    String consumerGroupId = Joiner.on("-").join(TOPIC, "auto", System.currentTimeMillis());
+    consumerProps.setProperty(SOURCE_KAFKA_CONSUMERCONFIG_KEY_WITH_DOT + HighLevelConsumer.GROUP_ID_KEY, consumerGroupId);
+    consumerProps.setProperty(HighLevelConsumer.ENABLE_AUTO_COMMIT_KEY, "true");
+
+    // Create an instance of MockedHighLevelConsumer using an anonymous class
+    MockedHighLevelConsumer consumer = new MockedHighLevelConsumer(TOPIC, ConfigUtils.propertiesToConfig(consumerProps), NUM_PARTITIONS) {
+      @Override
+      public void processMessage(DecodeableKafkaRecord<byte[], byte[]> message) {
+        super.processMessage(message);
+        // Override the method to throw a custom exception
+        throw new RuntimeException("Simulated exception in processMessage");
+      }
+    };
+    consumer.startAsync().awaitRunning();
+
+    // assert all NUM_MSGS messages were processed.
+    consumer.awaitExactlyNMessages(NUM_MSGS, 10000);
```

Comment:
I am not sure if this method tests the change. iiuc this method would have worked without the change also as `processMessage` adds to the queue before throwing exception and this method call just checks the queue size.

Also, add the test with `enable.auto.commit: false` and verify that exception doesn't cause processing to stop and that the same message is retried to validate that the message is not skipped.

Reply:
I tried running this test without the change in PR, it failed with timeout waiting for 10 messages

![image](https://github.com/user-attachments/assets/6d04e8c2-26f6-4fc6-8015-0a5c20c70bbb)

The reason is we want to process the message even if an exception is encountered. In older world the `processMessage` gets called only once. The loop breaks as soon as an exception is encountered.

Reply:
Updated the test with `enable.auto.commit: false` and asserted on no message getting committed.

Reply:
as mentioned in the other comment, this verifies that `NUM_MSGS` are seen, but each is seen only once (when it fails) and is never retried.  that is, it's skipped.

e.g. if you were to alternate to have the mocked `processMessage` throw an exception every other time, I believe you'd count N calls to `processMessage` and the commit offsets would increase by `NUM_MSGS`, but to demonstrate NO SKIPPING, you'd want `processMessage` to be called either `(2 * NUM_MSGS)` or `((2 * NUM_MSGS) - 1)` (depending on whether failure on odd or even invocations)

Reply:
Updated the PR to fix it only when `enable.auto.commit` is `true`.